lib/AST/ASTDumper.cpp

```diff
@@ -2118,15 +2118,26 @@ class PrintExpr : public ExprVisitor<PrintExpr> {
     PrintWithColorRAII(OS, ParenthesisColor) << ')';
   }
   void visitTupleShuffleExpr(TupleShuffleExpr *E) {
+    printCommon(E, "tuple_shuffle_expr");
+    OS << " elements=[";
+    for (unsigned i = 0, e = E->getElementMapping().size(); i != e; ++i) {
+      if (i) OS << ", ";
+      OS << E->getElementMapping()[i];
+    }
+    OS << "]\n";
+    printRec(E->getSubExpr());
+    PrintWithColorRAII(OS, ParenthesisColor) << ')';
+  }
+  void visitArgumentShuffleExpr(ArgumentShuffleExpr *E) {
     printCommon(E, "tuple_shuffle_expr");
```

Comment:
Did you mean to also update this to `argument_shuffle_expr`?

Reply:
Good catch. I grepped for TupleShuffleExpr but forgot to check other spellings. I'll fix this in a follow-up PR.